CMakeLists.txt

```diff
@@ -153,14 +159,15 @@ if(APPLE)
     set(XCODE_DEVELOPMENT_TEAM "" CACHE STRING "Xcode development team ID")
     set(PRODUCTBUILD_IDENTITY_NAME "" CACHE STRING "productbuild identity name")
     set(PRODUCTBUILD_KEYCHAIN_PATH "" CACHE FILEPATH "pkgbuild keychain file")
-    if(IOS)
+    if(APPLE_VAR_OS)
         set(XCODE_PROVISIONING_PROFILE_SPECIFIER "" CACHE STRING "Xcode provisioning profile specifier")
         set(MOLTENVK_SDK "$ENV{VULKAN_SDK}/../MoltenVK" CACHE STRING "MoltenVK directory inside the Vulkan SDK")
     endif()
 
     # Deployment
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.11" CACHE STRING "macOS Deployment Target")
-    if(IOS)
+    if(APPLE_MAC_OS)
+        set(CMAKE_OSX_DEPLOYMENT_TARGET "10.11" CACHE STRING "macOS Deployment Target")
+    elseif(IOS)
```

Comment:
What about setting deployment targets for tvOS and visionOS? And shouldn't we also set `CMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH` for those?

Reply:
Please answer this question.

Reply:
Makes sense, as we want to support the widest range of versions/devices possible.

`11.0` seemed like a good choice for tvOS as well, as it works with Xcode 14 and newer and all iPhones that have armv8 CPUs.

For visionOS I'd set it to `1.0`, so all versions are and will be included.

commit incoming.

Reply:
ONLY_ACTIVE_ARCH only makes a difference for tvOS simulator, so I'll include it.

For visionOS (both device and simulator SDK) only arm64 is supported, so I'll omit it there.